File: caffeine/src/main/java/project/study/caffeine/core/domain/city/entity/District.java
package project.study.caffeine.core.domain.city.entity;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import java.util.Objects;

@Entity(name = "district")
public class District {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long districtId;

    @Column
    private String name;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "cityId")
    private City city;

    public Long getDistrictId() {
        return districtId;
    }

    public City getCity() {
        return city;
    }

    public String getName() {
        return name;
    }

    public String getCityName() {
        return city.getName();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof District district)) return false;
        return getDistrictId().equals(district.getDistrictId());
    }

    @Override
    public int hashCode() {
        return Objects.hash(getDistrictId());
    }

    @Override
    public String toString() {
        return districtId.toString();
    }
}

File: jpa/jpa-basic/src/main/kotlin/project/study/jpa/core/domain/city/entity/District.kt
package project.study.jpa.core.domain.city.entity

import jakarta.persistence.*

@Entity
class District(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private var districtId: Long? = null,

    @Column
    private var name: String,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "cityId")
    private var city: City
) {
    fun getDistrictId(): Long? {
        return districtId
    }

    fun getName(): String {
        return name;
    }

    fun getCityName(): String {
        return city.getName()
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is District) return false
        if (districtId != other.districtId) return false
        return true
    }

    override fun hashCode(): Int {
        return districtId?.hashCode() ?: 0
    }

    override fun toString(): String {
        return districtId.toString()
    }
}
